tools/cldr-apps/src/main/java/org/unicode/cldr/web/VettingViewerQueue.java

```diff
@@ -725,6 +751,227 @@ public void writeVettingViewerOutput(CLDRLocale locale, StringBuffer aBuffer, We
+         * technical committee by committing directly to version control rather than voting.
+         */
+        CLDRFile sourceFile = sourceFactory.make(loc, true);
+        Factory baselineFactory = CLDRConfig.getInstance().getCommonAndSeedAndMainAndAnnotationsFactory();
+        CLDRFile baselineFile = baselineFactory.make(loc, true);
+        Relation<R2<SectionId, PageId>, VettingViewer<Organization>.WritingInfo> file;
+        file = vv.generateFileInfoReview(choiceSet, loc, usersOrg, usersLevel, quick, sourceFile, quick ? null : baselineFile);
+        return getReviewOutput(sourceFile, baselineFile, file, choiceSet, locale);
+    }
+
+    @Schema(description = "Heading for a portion of the notifications")
+    public static final class ReviewNotification{
+
+        public String notification;
+
+        public ReviewNotification(String notificationName) {
+            this.notification = notificationName;
+        }
+
+        // for serialization
+        public Object[] getEntries() {
+            return entries.toArray();
+        }
+
+        private List<ReviewNotificationGroup> entries = new ArrayList<>();
+
+        ReviewNotificationGroup add(ReviewNotificationGroup group) {
+            this.entries.add(group);
+            return group;
+        }
+    }
+
+    public static final class ReviewNotificationGroup {
+        private List<ReviewEntry> entries = new ArrayList<>();
+
+        public String section;
+        public String page;
+        public String header;
+
+        public ReviewNotificationGroup(String sectionName, String pageName, String headerName) {
+            this.section = sectionName;
+            this.page = pageName;
+            this.header = headerName;
+        }
+
+        public ReviewEntry add(String code, String xpath) {
+            ReviewEntry e = new ReviewEntry(code, xpath);
+            this.entries.add(e);
+            return e;
+        }
+
+        // for serialization
+        public Object[] getEntries() {
+            return entries.toArray();
+        }
+    }
+
+    public static final class ReviewOutput {
+        private List<ReviewNotification> notifications = new ArrayList<>();
+
+        public Object[] getNotifications() {
+            return notifications.toArray();
+        }
+
+        public ReviewNotification add(String notification) {
+            return add(new ReviewNotification(notification));
+        }
+
+        public ReviewNotification add(ReviewNotification notification) {
+            this.notifications.add(notification);
+            return notification;
+        }
+    }
+
+    /**
+     * Get Vetting Viewer Review output as an object
+     * @param sourceFile
+     * @param baselineFile
+     * @param sorted
+     * @param choices
+     * @param locale
+     * @return
+     */
+    private ReviewOutput getReviewOutput(CLDRFile sourceFile, CLDRFile baselineFile,
+        Relation<R2<SectionId, PageId>, VettingViewer<Organization>.WritingInfo> sorted,
```

Comment:
this function is too long, would benefit from subroutines

Reply:
done